src/main/kotlin/org/rust/ide/formatter/impl/spacing.kt

```diff
@@ -138,6 +138,7 @@ fun createSpacingBuilder(commonSettings: CommonCodeStyleSettings, rustSettings:
 
         //== rules with very large area of application
         .around(NO_SPACE_AROUND_OPS).spaces(0)
+        .around(BINARY_OP).spaces(1)
```

Comment:
I'm not very sure that it's the best fix but tests work:)

Reply:
I think this rule might duplicate the rule below

Reply:
Yeah, but we can't just exclude `RS_BINARY_OPS` tokens from `SPACE_AROUND_OPS` because, for example, `=` and `|` tokens can be part not only binary expressions.
We can exclude all `RS_BINARY_OPS` tokens except `=` and `|` but I'm afraid of that there are other cases which are not in our tests.

Reply:
If you know a good way to get rid of this duplication, I will be glad to use it